src/main.cpp

```diff
@@ -4991,12 +4991,12 @@ bool LoadExternalBlockFile(const CChainParams &chainparams, FILE *fileIn, CDiskB
             {
                 // locate a header
                 unsigned char buf[MESSAGE_START_SIZE];
-                blkdat.FindByte(chainparams.MessageStart()[0]);
+                blkdat.FindByte(chainparams.MessageStart()[0]); // FIXME do we have to use GetMagic?
                 // FindByte peeks 1 ahead and locates the file pointer AT the byte, not at the next one as is typical
                 // for file ops.  So if we rewind, we want to go one further.
                 nRewind = blkdat.GetPos() + 1;
                 blkdat >> FLATDATA(buf);
-                if (memcmp(buf, chainparams.MessageStart(), MESSAGE_START_SIZE))
+                if (memcmp(buf, chainparams.MessageStart(), MESSAGE_START_SIZE)) // FIXME do we have to use GetMagic?
                     continue;
                 // read size
```

Comment:
I would think we don't want to add any new magic here for reading/writing to disk.  Did you try a reindex? It should work just fine as is.

Reply:
yeah this reason why I added it as with a question mark (and also I gorgot to mark the PR as WIP)